pkg/distributor/ha_tracker.go

```diff
@@ -274,6 +283,41 @@ const (
 	deletionTimeout = 30 * time.Minute
 )
 
+func (h *haTracker) processKVStoreEntry(key string, replica *ReplicaDesc) bool {
+	segments := strings.SplitN(key, "/", 2)
+
+	// Valid key would look like cluster/replica, and a key without a / such as `ring` would be invalid.
+	if len(segments) != 2 {
+		return true
+	}
+
+	user := segments[0]
+	cluster := segments[1]
+
+	if replica.DeletedAt > 0 {
+		h.electedReplicaChanges.DeleteLabelValues(user, cluster)
+		h.electedReplicaTimestamp.DeleteLabelValues(user, cluster)
+
+		h.electedLock.Lock()
+		defer h.electedLock.Unlock()
+		userClusters := h.clusters[user]
+		if userClusters != nil {
+			delete(userClusters, cluster)
+			if len(userClusters) == 0 {
+				delete(h.clusters, user)
+			}
+		}
+		return true
+	}
+
+	// Store the received information into our cache
+	h.electedLock.Lock()
+	h.updateCache(user, cluster, replica)
+	h.electedLock.Unlock()
```

Comment:
I have a question, just for knowledge learning. for the `WatchPrefix` 
Comments say 
```
 Request callbacks from KVStore when data changes
```
But i'm curious how important this is as it seems to be only called once instead of staying open for any changes for N amount of time time we do for ticker's for `updateKVLoop`.

I would of thought 
`WatchPrefix` will stay open as long as this code below is still going.
```
go func() {
		defer wg.Done()
		h.updateKVLoop(ctx)
	}()
```
 Since this will do a ticker and update the KVStore or clean up Replicas. And I would of thought `WatchPrefix` would watch for those changes each time?